CMake/Utilities.cmake

```diff
@@ -29,3 +33,41 @@ function(IN_STR_LIST LIST_NAME ITEM_SEARCHED RETVAL)
     set(${RETVAL} TRUE PARENT_SCOPE)
   endif()
 endfunction()
+
+# Set Compiler flags
+function(SET_BUILD_TYPE_FLAGS BUILD_TYPE)
+  if(WIN32)
+    set(CompilerFlags
+        CMAKE_C_FLAGS
+        CMAKE_C_FLAGS_DEBUG
+        CMAKE_C_FLAGS_MINSIZEREL
+        CMAKE_C_FLAGS_RELEASE
+        CMAKE_C_FLAGS_RELMITDEBINFO
+        CMAKE_CXX_FLAGS
+        CMAKE_CXX_FLAGS_DEBUG
+        CMAKE_CXX_FLAGS_MINSIZEREL
+        CMAKE_CXX_FLAGS_RELEASE
+        CMAKE_CXX_FLAGS_RELMITDEBINFO)
+
+    #Set /MT flag for Static compiling, /MD for Shared
+    foreach(Flag ${CompilerFlags})
+      if("${BUILD_TYPE}" MATCHES "STATIC")
+        string(REPLACE "/MD" "/MT" ${Flag} "${${CompilerFlags}}")
+      else()
+        string(REPLACE "/MT" "/MD" ${Flag} "${${CompilerFlags}}")
+      endif()
+    endforeach()
+  else()
+    if("${BUILD_TYPE}" MATCHES "STATIC")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
```

Comment:
Why do you exactly have to set fPIC for static builds? I think you inversed the logic here. All these flags should not be needed if you use `add_library(... STATIC ...)`.